src/utils/keyHelpers/acl.ts

```diff
@@ -0,0 +1,74 @@
+import * as debug from '../../debug'
+import { store } from 'solid-logic'
+
+/**
+ * set ACL
+ * @param keyDoc
+ * @param aclBody
+ */
+export async function setAcl (keyDoc: string, aclBody: string) {
+  // Some servers don't present a Link http response header
+  // if the container doesn't exist yet, refetch the resource
+
+  await store.fetcher.load(keyDoc)
+
+  // FIXME: check the Why value on this quad:
+  debug.log(store.statementsMatching(store.sym(keyDoc), store.sym('http://www.iana.org/assignments/link-relations/acl')))
+  const keyAclDoc = store.any(store.sym(keyDoc), store.sym('http://www.iana.org/assignments/link-relations/acl'))
+  if (!keyAclDoc) {
+    throw new Error('Key ACL doc not found!')
+  }
+
+  try {
+    await store.fetcher.webOperation('PUT', keyAclDoc.value, {
+      data: aclBody,
+      contentType: 'text/turtle'
+    })
+  } catch (err) {
+    if (err?.response?.status !== 404) { throw new Error(err) }
```

Comment:
@bourgeoa I'm wondering, if it errors with a 404, should we create the resource in that case?

And otherwise if it errors would it be because the user is not logged in? If so I will mock that data for testing.

Reply:
PUT should never error with 404 so we should remove the if
The reason for failing could be 401 or 403 (not authenticated or not authorized)